app/models/match.rb

```diff
@@ -390,6 +392,19 @@ def update_user_stats_constructed
     end
   end
 
+  def ==(match)
+    self.oppname == match.oppname && self.duration == match.duration
+  end
+
+  ### VALIDATION METHODS:
+
+  def no_duplicate_matches
+    user = User.find(self.user_id)
```

Comment:
Every match created now has to look up the user again. I wonder if we can skip this in some way?

Does the dupes come in a row or in separate times?